src/werkzeug/debug/shared/debugger.js

```diff
@@ -1,210 +1,319 @@
 });
 
 function initPinBox() {
-  $('.pin-prompt form').submit(function(evt) {
-    evt.preventDefault();
-    var pin = this.pin.value;
-    var btn = this.btn;
-    btn.disabled = true;
-    $.ajax({
-      dataType: 'json',
-      url: document.location.pathname,
-      data: {__debugger__: 'yes', cmd: 'pinauth', pin: pin,
-             s: SECRET},
-      success: function(data) {
-        btn.disabled = false;
-        if (data.auth) {
-          EVALEX_TRUSTED = true;
-          $('.pin-prompt').fadeOut();
-        } else {
-          if (data.exhausted) {
-            alert('Error: too many attempts.  Restart server to retry.');
-          } else {
-            alert('Error: incorrect pin');
-          }
-        }
-        console.log(data);
-      },
-      error: function() {
-        btn.disabled = false;
-        alert('Error: Could not verify PIN.  Network error?');
-      }
-    });
-  });
+    document.querySelector(".pin-prompt form")
+        .addEventListener("submit", function(event) {
+            event.preventDefault();
+            const pin = encodeURIComponent(this.pin.value);
+            const encodedSecret = encodeURIComponent(SECRET);
+            const btn = this.btn;
+            btn.disabled = true;
+
+            fetch(`${document.location.pathname}?__debugger__=yes&cmd=pinauth&pin=${pin}&s=${encodedSecret}`)
+                .then(res => res.json())
+                .then(({ auth, exhausted }) => {
+                    if (auth) {
+                        EVALEX_TRUSTED = true;
+                        fadeOut(document.getElementsByClassName("pin-prompt")[0]);
+                    } else {
+                        alert(`Error: ${exhausted
+                        ? "too many attempts.  Restart server to retry."
+                        : "incorrect pin"}`)
+                    }
+                })
+                .catch(err => {
+                    alert('Error: Could not verify PIN.  Network error?');
+                    console.error(err);
+                })
+                .finally(() => btn.disabled = false);
+        }, false);
 }
 
 function promptForPin() {
-  if (!EVALEX_TRUSTED) {
-    $.ajax({
-      url: document.location.pathname,
-      data: {__debugger__: 'yes', cmd: 'printpin', s: SECRET}
+    if (!EVALEX_TRUSTED) {
+        const encodedSecret = encodeURIComponent(SECRET);
+        fetch(`${document.location.pathname}?__debugger__=yes&cmd=printpin&s=${encodedSecret}`);
+        const pinPrompt = document.getElementsByClassName("pin-prompt")[0];
+        fadeIn(pinPrompt);
+        document.querySelector('.pin-prompt input[name="pin"]').focus();
+    }
+}
+
+/**
+ * Helper function for shell initialization
+ */
+function openShell(consoleNode, target, frameID) {
+    promptForPin();
+    if (consoleNode) {
+        slideToggle(consoleNode);
+        return consoleNode;
+    }
+
+    let historyPos = 0;
+    let history = [''];
```

Comment:
that one can be `const` since const only prevents reassignment (`foo = bar`) but not mutation (`foo.push()` etc)